### ISSUE TITLE
Pod config dropping value when using :[number][alpha] syntax

### ISSUE THREAD

**Issue by JJ** [rakudo/rakudo#2793]
## The Problem

This program:

```perl6
=begin pod :3foo
Does not really matter
=end pod

=begin pod :foo(3)
Does not really matter much
=end pod

dd $=pod;
```

## Expected Behavior

Should have the same value for config, since :3foo === :foo(3)

## Actual Behavior

, but it prints

```perl6
Array element = [Pod::Block::Named.new(name => "pod", config => {:foo}, contents => [Pod::Block::Para.new(config => {}, contents => ["Does not really matter"])]), Pod::Block::Named.new(name => "pod", config => {:foo(3)}, contents => [Pod::Block::Para.new(config => {}, contents => ["Does not really matter much"])])]
```

In the case of `:3foo`, the 3 is dropped and the value turned into a Bool

## Environment
* Operating system: Ubuntu 16.04
* Compiler version (`perl6 -v`): 

```
This is Rakudo version 2019.03.1 built on MoarVM version 2019.03
implementing Perl 6.d
```

**Comment by tbrowder** [comment]
I don't see that sugar method mentioned in S26, so I didn't think of implementing it, but it should be possible.

**Comment by JJ** [comment]
Well, if only for consistency... If documented, we can just drop it.

**Comment by tbrowder** [comment]
Hm, mark it NYI?

**Comment by JJ** [comment]
Maybe. Come to think of it, the problem with that syntax is that it's erroneously processed, with :3whatever exactly equal than :whatever. So it should either fail as NYI or be implemented, whatever is the easiest.

**Comment by tbrowder** [comment]
If you mean throw some kind of NYI error, I would have as much trouble with that as with implementing it!  So I might as well take it on.

**Comment by tbrowder** [comment]
After looking at a two-digit test (e.g., "=begin pod :34foo") using module Data::Dump for more details on the current structure produced by rakudo, I believe the fix is relatively straight forward by:

+ modifying the boolean (key = 'numbered') path in sub make_config in "src/Perl6/Pod.nqp"

**Comment by JJ** [comment]
Great!

**Comment by tbrowder** [comment]
Closed with 0d47db8.

**Comment by tbrowder** [comment]
By the way, the "number" must be a UInt, and the docs should reflect that but I don't think that's true.
